buildingspy/development/validator.py

```diff
@@ -177,3 +179,377 @@ def _validateHTML(self, moFile):
+            elif(float(value) == 0.0):
+                s = ("Found mos file=" + str(mos_file) + " without tolerance specified.\n" + 
+                    "A maximum tolerance of 1e-6 is required by JModelica for unit tests.\n")
+                raise ValueError(s)
+        if ("" + name + "=" + "" == "stopTime="):
+            if(float(value) == 0.0):
+                s = ("Found mos file=" + str(mos_file) + " without stopTime specified.\n" + 
+                    "A non-null stopTime is required by OpenModelica for unit tests.\n")
+                raise ValueError(s)
+                
+    def _wrong_literal (self, mos_file, name):
+        """ 
+        Stop if invalid literal is found.
+    
+        :param mos_file: mos file.
+        :param name: Parameter name.
+    
+         """
+        
+        s = ("Found mos file=" + str(mos_file) + " with invalid expression=" 
+             + str(name+'='+name) + ".\n" 
+             + "This is not allowed for cross validation with JModelica.\n")
+        raise ValueError(s)
+    
+    
+    def _validate_model_parameters (self, name, mos_files, root_dir):
+        """ 
+        Validate parameter settings.
+    
+        :param name: Parameter name.
+        :param mos_files: List of mos files.
+        :param root_dir: Root directory.
+    
+         """
+
+        N_mos_defect=0
+    
+        j = 1
+        for mos_file in mos_files:
+            j += 1
+            
+            f = open(mos_file,"r", encoding="utf8")
+            
+            content = f.readlines()
+            found = False
+            i = 0
+            while found == False and i<len(content):
+                l = content[i]
+                if "simulateModel(" in l.replace(" ", ""):
+                    line = l
+                    found = True
+                i += 1
+            
+            try:
+                if ""+name+"="+name+"" in line.replace(" ", ""):
+                    value = ""+name+""
+                    self._wrong_literal(mos_file ,name)
+                   
+                if ""+name+"="+"" in line.replace(" ", ""):
+                    pTime    = re.compile(r"[\d\S\s.,]*("+name+"=)([\d]*[.]*[\d]*[eE]*[+|-]*[\d]*[*]*[\d]*[.]*[\d]*[eE]*[+|-]*[\d]*)")
+                    mTime    = pTime.match(line)
+                    value = mTime.group(2)
+                    self._check_tolerance(content, name, value, mos_file)     
+                else:
+                    found = False
+                    while found == False and i<len(content):
+                        line = content[i]
+                        i += 1
+                        # Remove white spaces
+                        line.replace(" ", "")
+                        
+                        if ""+name+"="+"" in line.replace(" ", ""):
+                            found = True
+                            pTime    = re.compile(r"[\d\S\s.,]*("+name+"=)([\d]*[.]*[\d]*[eE]*[+|-]*[\d]*[*]*[\d]*[.]*[\d]*[eE]*[+|-]*[\d]*)[\S\s.,]*")
```

Comment:
Is this `pTime` reg expression the same as on line 430? It seems a bit different, but I don't see why and what it is doing as there is no comment and otherwise the code seems to be the same as around line 430. Please correct to be clearer (and give others a chance to edit your code if needed).